### DISCUSSION TITLE
esphome use esp32-ble-example.yaml boot stop

### DISCUSSION THREAD

**Discussion by Wilsonqwx** [syssi/esphome-jk-bms#191] [Q&A] [answered]
Hello All,
trying to get esp32 to work with my jk-B2a20s20p, use home assistant ESPHome to flash esp32-ble-example.yaml
log as below

ets Jul 29 2019 12:21:46

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:7548
ho 0 tail 12 room 4
load:0x40078000,len:14912
load:0x40080400,len:3688
entry 0x4008067c
I (29) boot: ESP-IDF 4.3.2 2nd stage bootloader
I (29) boot: compile time 22:24:13
I (29) boot: chip revision: 3
I (32) boot_comm: chip revision: 3, min. bootloader chip revision: 0
I (39) boot.esp32: SPI Speed      : 40MHz
I (44) boot.esp32: SPI Mode       : DIO
I (48) boot.esp32: SPI Flash Size : 4MB
I (53) boot: Enabling RNG early entropy source...
I (58) boot: Partition Table:
I (62) boot: ## Label            Usage          Type ST Offset   Length
I (69) boot:  0 nvs              WiFi data        01 02 00009000 00004000
I (76) boot:  1 otadata          OTA data         01 00 0000d000 00002000
I (84) boot:  2 phy_init         RF data          01 01 0000f000 00001000
I (91) boot:  3 app0             OTA app          00 10 00010000 001c0000
I (99) boot:  4 app1             OTA app          00 11 001d0000 001c0000
I (106) boot: End of partition table
I (111) boot_comm: chip revision: 3, min. application chip revision: 0
I (118) esp_image: segment 0: paddr=00010020 vaddr=3f400020 size=39540h (234816) map
I (211) esp_image: segment 1: paddr=00049568 vaddr=3ffbdb60 size=04928h ( 18728) load
I (219) esp_image: segment 2: paddr=0004de98 vaddr=40080000 size=02180h (  8576) load
I (223) esp_image: segment 3: paddr=00050020 vaddr=400d0020 size=eaaa0h (961184) map
I (573) esp_image: segment 4: paddr=0013aac8 vaddr=40082180 size=1aab4h (109236) load
I (618) esp_image: segment 5: paddr=00155584 vaddr=50000000 size=00010h (    16) load
I (633) boot: Loaded app from partition at offset 0x10000
I (633) boot: Disabling RNG early entropy source...
I (644) cpu_start: Pro cpu up.
I (644) cpu_start: Starting app cpu, entry point is 0x40081da0
I (0) cpu_start: App cpu up.
I (661) cpu_start: Pro cpu start user code
I (661) cpu_start: cpu freq: 160000000
I (661) cpu_start: Application information:
I (665) cpu_start: Project name:     jkbmsble
I (670) cpu_start: App version:      2022.9.4
I (675) cpu_start: Compile time:     Oct 15 2022 22:23:27
I (681) cpu_start: ELF file SHA256:  7449790451549ac8...
I (687) cpu_start: ESP-IDF:          4.3.2
I (693) heap_init: Initializing. RAM available for dynamic allocation:
I (699) heap_init: At 3FFAFF10 len 000000F0 (0 KiB): DRAM
I (706) heap_init: At 3FFB6388 len 00001C78 (7 KiB): DRAM
I (712) heap_init: At 3FFB9A20 len 00004108 (16 KiB): DRAM
I (718) heap_init: At 3FFCA868 len 00015798 (85 KiB): DRAM
I (724) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (730) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM
I (737) heap_init: At 4009CC34 len 000033CC (12 KiB): IRAM
I (744) spi_flash: detected chip: generic
I (748) spi_flash: flash io: dio
I (753) cpu_start: Starting scheduler on PRO CPU.
I (0) cpu_start: Starting scheduler on APP CPU.
can you please help me to figure out the problem.

**Comment by Wilsonqwx** [comment, marked as answer]
after press the en bottom on the esp32 board, it generated following logs

[I][logger:243]: Log initialized
[VV][esp32.preferences:070]: nvs_get_blob: key: 233825507, len: 4
[C][ota:465]: There have been 6 suspected unsuccessful boot attempts.
[VV][esp32.preferences:039]: s_pending_save: key: 233825507, len: 4
[D][esp32.preferences:113]: Saving 1 preferences to flash...
[VV][esp32.preferences:122]: Checking if NVS data 233825507 has changed
[V][esp32.preferences:125]: sync: key: 233825507, len: 4
[D][esp32.preferences:142]: Saving 1 preferences to flash: 0 cached, 1 written, 0 failed
[I][app:029]: Running through setup()...
[V][app:030]: Sorting components by setup priority...
[VV][esp-idf:000]: I (62) BTDM_INIT: BT controller compile version [6a07b06]

[VV][esp-idf:000]: I (901) system_api: Base MAC address is not set

[VV][esp-idf:000]: I (902) system_api: read default base MAC address from EFUSE

[VV][esp-idf:000]: I (913) phy_init: phy_version 4670,719f9f6,Feb 18 2021,17:07:07

[D][esp32_ble_tracker:264]: Starting scan...
[C][wifi:037]: Setting up WiFi...
[VV][esp-idf:000]: I (1531) wifi:
[VV][esp-idf:000]: wifi driver task: 3ffdfea0, prio:23, stack:6656, core=0
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (1536) wifi:
[VV][esp-idf:000]: wifi firmware version: eb52264
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (1543) wifi:
[VV][esp-idf:000]: wifi certification version: v7.0
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (1544) wifi:
[VV][esp-idf:000]: config NVS flash: enabled
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (1554) wifi:
[VV][esp-idf:000]: config nano formating: disabled
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (1575) wifi:
[VV][esp-idf:000]: Init data frame dynamic rx buffer num: 32
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (1585) wifi:
[VV][esp-idf:000]: Init management frame dynamic rx buffer num: 32
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (1605) wifi:
[VV][esp-idf:000]: Init management short buffer num: 32
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (1615) wifi:
[VV][esp-idf:000]: Init dynamic tx buffer num: 32
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (1626) wifi:
[VV][esp-idf:000]: Init static rx buffer size: 1600
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (1636) wifi:
[VV][esp-idf:000]: Init static rx buffer num: 10
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (1646) wifi:
[VV][esp-idf:000]: Init dynamic rx buffer num: 32
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (1657) wifi_init: rx ba win: 6

[VV][esp-idf:000]: I (1666) wifi_init: tcpip mbox: 32

[VV][esp-idf:000]: I (1667) wifi_init: udp mbox: 6

[VV][esp-idf:000]: I (1677) wifi_init: tcp mbox: 6

[VV][esp-idf:000]: I (1687) wifi_init: tcp tx win: 5744

[VV][esp-idf:000]: I (1687) wifi_init: tcp rx win: 5744

[VV][esp-idf:000]: I (1697) wifi_init: tcp mss: 1440

[VV][esp-idf:000]: I (1707) wifi_init: WiFi IRAM OP enabled

[VV][esp-idf:000]: I (1708) wifi_init: WiFi RX IRAM OP enabled

[V][esp32.preferences:058]: nvs_get_blob('2677304903'): ESP_ERR_NVS_NOT_FOUND - the key might not be set yet
[V][wifi_esp32:202]: Enabling STA.
[VV][esp-idf:000]: I (1731) wifi:
[VV][esp-idf:000]: mode : sta (78:21:84:9a:ee:04)
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (1749) wifi:
[VV][esp-idf:000]: enable tsf
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (1759) wifi:
[VV][esp-idf:000]: Set ps type: 1

[VV][esp-idf:000]: 

[D][wifi:384]: Starting scan...
[VV][esp-idf:000]: W (1770) wifi:
[VV][esp-idf:000]: Error! Should use default active scan time parameter for WiFi scan when Bluetooth is enabled!!!!!!

[VV][esp-idf:000]: 

[W][jk_bms_ble:342]: [c8:47:8c:e1:e2:e1] Not connected
[V][wifi_esp32:591]: Event: WiFi STA start
[VV][esp-idf:000]: I (1800) wifi:
[VV][esp-idf:000]: Set ps type: 1

[VV][esp-idf:000]: 

[V][esp32_ble_client:078]: gattc registered app id 1
[D][switch:037]: 'jkbmsble enable bluetooth connection': Sending state ON
[D][esp32_ble_tracker:803]: Found device 06:1E:DF:FB:EB:AE RSSI=-89
[D][esp32_ble_tracker:824]:   Address Type: RANDOM
[V][wifi_esp32:659]: Event: WiFi Scan Done status=0 number=6 scan_id=128
[D][wifi:399]: Found networks:
[I][wifi:443]: - 'Ruckus' [redacted]▂▄▆█
[D][wifi:444]:     Channel: 8
[D][wifi:445]:     RSSI: -88 dB
[I][wifi:443]: - 'Ruckus' [redacted]▂▄▆█
[D][wifi:444]:     Channel: 8
[D][wifi:445]:     RSSI: -95 dB
[D][wifi:448]: - [redacted] [redacted]▂▄▆█
[D][wifi:448]: - [redacted] [redacted]▂▄▆█
[D][wifi:448]: - [redacted] [redacted]▂▄▆█
[D][wifi:448]: - [redacted] [redacted]▂▄▆█
[I][wifi:255]: WiFi Connecting to 'Ruckus'...
[V][wifi:257]: Connection Params:
[V][wifi:258]:   SSID: 'Ruckus'
[V][wifi:261]:   BSSID: 1A:59:C0:33:89:13
[V][wifi:281]:   Password: [redacted]
[V][wifi:286]:   Channel: 8
[V][wifi:295]:   Using DHCP IP
[V][wifi:297]:   Hidden: NO
[VV][esp-idf:000]: I (5061) wifi:
[VV][esp-idf:000]: new:<8,0>, old:<1,0>, ap:<255,255>, sta:<8,0>, prof:1
[VV][esp-idf:000]: 

[W][jk_bms_ble:342]: [c8:47:8c:e1:e2:e1] Not connected
[VV][esp-idf:000]: I (5833) wifi:
[VV][esp-idf:000]: state: init -> auth (b0)
[VV][esp-idf:000]: 

[D][esp32_ble_tracker:803]: Found device 5C:6C:61:A8:C8:4F RSSI=-74
[D][esp32_ble_tracker:824]:   Address Type: RANDOM
[D][esp32_ble_tracker:828]:   TX Power: 2
[VV][esp-idf:000]: I (5862) wifi:
[VV][esp-idf:000]: state: auth -> assoc (0)
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (6865) wifi:
[VV][esp-idf:000]: state: assoc -> init (400)
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (6867) wifi:
[VV][esp-idf:000]: new:<8,0>, old:<8,0>, ap:<255,255>, sta:<8,0>, prof:1
[VV][esp-idf:000]: 

[W][wifi_esp32:633]: Event: Disconnected ssid='Ruckus' bssid=[redacted] reason='Association Expired'
[W][wifi:547]: Error while connecting to network.
[W][wifi:583]: Restarting WiFi adapter...
[V][wifi_esp32:204]: Disabling STA.
[VV][esp-idf:000]: I (6900) wifi:
[VV][esp-idf:000]: flush txq
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (6901) wifi:
[VV][esp-idf:000]: stop sw txq
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (6911) wifi:
[VV][esp-idf:000]: lmac stop hw txq
[VV][esp-idf:000]: 

[I][wifi:255]: WiFi Connecting to 'Ruckus'...
[V][wifi:257]: Connection Params:
[V][wifi:258]:   SSID: 'Ruckus'
[V][wifi:261]:   BSSID: 1A:59:C0:33:89:13
[V][wifi:281]:   Password: [redacted]
[V][wifi:286]:   Channel: 8
[V][wifi:295]:   Using DHCP IP
[V][wifi:297]:   Hidden: NO
[V][wifi_esp32:202]: Enabling STA.
[VV][esp-idf:000]: I (7038) wifi:
[VV][esp-idf:000]: mode : sta (78:21:84:9a:ee:04)
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (7045) wifi:
[VV][esp-idf:000]: enable tsf
[VV][esp-idf:000]: 

[D][esp32_ble_tracker:803]: Found device 2B:BB:4A:A8:AE:04 RSSI=-78
[D][esp32_ble_tracker:824]:   Address Type: RANDOM
[V][wifi_esp32:603]: Event: WiFi STA stop
[V][wifi_esp32:591]: Event: WiFi STA start
[VV][esp-idf:000]: I (7076) wifi:
[VV][esp-idf:000]: Set ps type: 1

[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (7109) wifi:
[VV][esp-idf:000]: new:<8,0>, old:<1,0>, ap:<255,255>, sta:<8,0>, prof:1
[VV][esp-idf:000]: 

[VV][esp-idf:000]: I (7111) wifi:
[VV][esp-idf:000]: state: init -> auth (b0)
[VV][esp-idf:000]: 

ets Jul 29 2019 12:21:46

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:7548
ho 0 tail 12 room 4
load:0x40078000,len:14912
load:0x40080400,len:3688
entry 0x4008067c
I (29) boot: ESP-IDF 4.3.2 2nd stage bootloader
I (29) boot: compile time 22:24:13
I (29) boot: chip revision: 3
I (32) boot_comm: chip revision: 3, min. bootloader chip revision: 0
I (39) boot.esp32: SPI Speed      : 40MHz
I (44) boot.esp32: SPI Mode       : DIO
I (48) boot.esp32: SPI Flash Size : 4MB
I (53) boot: Enabling RNG early entropy source...
I (58) boot: Partition Table:
I (62) boot: ## Label            Usage          Type ST Offset   Length
I (69) boot:  0 nvs              WiFi data        01 02 00009000 00004000
I (76) boot:  1 otadata          OTA data         01 00 0000d000 00002000
I (84) boot:  2 phy_init         RF data          01 01 0000f000 00001000
I (91) boot:  3 app0             OTA app          00 10 00010000 001c0000
I (99) boot:  4 app1             OTA app          00 11 001d0000 001c0000
I (106) boot: End of partition table
I (111) boot_comm: chip revision: 3, min. application chip revision: 0
I (118) esp_image: segment 0: paddr=00010020 vaddr=3f400020 size=39540h (234816) map
I (211) esp_image: segment 1: paddr=00049568 vaddr=3ffbdb60 size=04928h ( 18728) load
I (219) esp_image: segment 2: paddr=0004de98 vaddr=40080000 size=02180h (  8576) load
I (223) esp_image: segment 3: paddr=00050020 vaddr=400d0020 size=eaaa0h (961184) map
I (573) esp_image: segment 4: paddr=0013aac8 vaddr=40082180 size=1aab4h (109236) load
I (618) esp_image: segment 5: paddr=00155584 vaddr=50000000 size=00010h (    16) load
I (633) boot: Loaded app from partition at offset 0x10000
I (633) boot: Disabling RNG early entropy source...
I (644) cpu_start: Pro cpu up.
I (644) cpu_start: Starting app cpu, entry point is 0x40081da0
I (0) cpu_start: App cpu up.
I (661) cpu_start: Pro cpu start user code
I (661) cpu_start: cpu freq: 160000000
I (661) cpu_start: Application information:
I (665) cpu_start: Project name:     jkbmsble
I (670) cpu_start: App version:      2022.9.4
I (675) cpu_start: Compile time:     Oct 15 2022 22:23:27
I (681) cpu_start: ELF file SHA256:  7449790451549ac8...
I (687) cpu_start: ESP-IDF:          4.3.2
I (693) heap_init: Initializing. RAM available for dynamic allocation:
I (699) heap_init: At 3FFAFF10 len 000000F0 (0 KiB): DRAM
I (706) heap_init: At 3FFB6388 len 00001C78 (7 KiB): DRAM
I (712) heap_init: At 3FFB9A20 len 00004108 (16 KiB): DRAM
I (718) heap_init: At 3FFCA868 len 00015798 (85 KiB): DRAM
I (724) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (730) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM
I (737) heap_init: At 4009CC34 len 000033CC (12 KiB): IRAM
I (744) spi_flash: detected chip: generic
I (748) spi_flash: flash io: dio
I (753) cpu_start: Starting scheduler on PRO CPU.
I (0) cpu_start: Starting scheduler on APP CPU.

**Reply by syssi** [reply]
If you remove the external component does the ESPHome node boot properly? At the moment it looks like it gets stuck on boot up.

Some ideas:
1. Move the ESP32 next to your WiFi access point and try again.
2. Remove all `jk_bms_ble` sections (and sensors) from the configuration YAML. Just leave the `esphome`, `logger`, `wifi` and `api` sections and try again.
3. Use another ESP32. It looks like the model/type of this one isn't detected properly.

**Reply by Wilsonqwx** [reply]
thanks for the quick response. it looks like the esp32 board is faulty, I tried some very basic code from esphome.io and it have the same error and will not be able to connected to Wi-Fi.
